packages/template-retail-react-app/app/components/_app/index.jsx

```diff
@@ -153,7 +172,10 @@ const App = (props) => {
         <Box className="sf-app" {...styles.container}>
             {/* For pwa-kit v2, we need to use this getToken prop to make sure it can work properly
                 with Storefront Preview feature*/}
-            <StorefrontPreview getToken={() => window.localStorage.getItem('token')}>
+            <StorefrontPreview
+                getToken={() => window.localStorage.getItem('token')}
+                onInit={initStorefrontPreview}
+            >
```

Comment:
Now removing the coupling between the template and pwa-kit-react-sdk:
By passing in a callback, then it would be the template that defines the wiring of `apiClients.willSendRequest` -> this is better than what I had, where StorefrontPreview somehow knew about the inner workings of the api clients.